### ISSUE TITLE
Basic auth with token as a password fails to download build-info-extractor-maven3-2.32.0-uber.jar from local Artifactory

### ISSUE THREAD

**Issue by Hi-Fi** [jfrog/jfrog-cli#1454]
**Describe the bug**
When running Jfrog CLI in Ci pipeline to collect Maven dependency information, it fails when in basic (username and password) authentication password is replaced with token. According https://www.jfrog.com/confluence/display/JFROG/Access+Tokens#AccessTokens-BasicAuthentication password and token should be fully exchangeable in basic auth.

**To Reproduce**
- Have local Artifactory that also has mirrored `build-info-extractor-maven3-2.32.0-uber.jar`.
- Have some basic pom.xml with some dependencies in the directory
- Have environment variables set
  - Artifactory URL as `ARTIFACTORY_URL` 
  - Username to Artifactory as `LOGIN_USER`
  - Artifactory Identity token as `LOGIN_PASSWORD`

```
jfrog c add --url "${ARTIFACTORY_URL}" --user "${LOGIN_USER}" --password "${LOGIN_PASSWORD}" --basic-auth-only
jfrog mvn-config
jfrog mvn clean install -Dartifactory.publish.artifacts=false
```

This produces error:
```
[Info] maven build config successfully created.
[Info] Running Mvn...
[Info] The build-info-extractor jar is not cached locally. Downloading it now...
 You can set the repository from which this jar is downloaded. Read more about it at https://www.jfrog.com/confluence/display/CLI/CLI+for+JFrog+Artifactory#CLIforJFrogArtifactory-DownloadingtheMavenandGradleExtractorJARs
[Info] Downloading build-info-extractor from https://artifactory.example.com/artifactory/extractors/org/jfrog/buildinfo/build-info-extractor-maven3/2.32.0/build-info-extractor-maven3-2.32.0-uber.jar
[Error] 401  received when attempting to download https://artifactory.example.com/artifactory/extractors/org/jfrog/buildinfo/build-info-extractor-maven3/2.32.0/build-info-extractor-maven3-2.32.0-uber.jar
```

**Expected behavior**
build-info-extractor and package dependencies downloaded correctly.

**Versions**
- JFrog CLI version: 2.9.0
- JFrog CLI operating system: Linux (Docker: `maven:3-openjdk-11` image)
- Artifactory Version: 7.29.10

**Additional context**
If authentication is done as `jfrog c add --url "${ARTIFACTORY_URL}" --access-token "${LOGIN_TOKEN}"`, build-info-extractor gets downloaded fine.

**Comment by sverdlov93** [comment]
HI @Hi-Fi,
I believe that your issue can be solved by https://github.com/jfrog/jfrog-cli/issues/1616

**Comment by Hi-Fi** [comment]
> HI @Hi-Fi,
> I believe that your issue can be solved by https://github.com/jfrog/jfrog-cli/issues/1616

Yes, sounds like same thing. Then just have to wait for on-prem version and it's upgrade at our env to verify.

**Comment by sverdlov93** [comment]
@Hi-Fi 
Please notice the workaround shared here - https://github.com/jfrog/jfrog-cli/issues/1616#issuecomment-1217734448 for earlier Artifactory versions.
